src/pages/preferences/preferencesTabs/publicKeysTab/EditModal.tsx

```diff
@@ -197,7 +195,7 @@ const ErrorContainer = styled.div`
 const KEY_NAME_ID = "key-name-input";
 const KEY_VALUE_ID = "key-value-input";
 const DUPLICATE_KEY_NAME = "The key name already exists.";
-const INVALID_SSH_KEY = "The SSH key must begin with 'ssh-rsa' or 'ssh-dss'.";
+const INVALID_SSH_KEY =
+  "The SSH key must begin with 'ssh-rsa' or 'ssh-dss' or 'ssh-ed25519' or 'ecdsa-sha2-nistp256'.";
 const EMPTY_KEY_NAME = "The key name cannot be empty.";
-const SSH_RSA = "ssh-rsa";
-const SSH_DSS = "ssh-dss";
+const VALID_SSH_KEY = /^(ssh-rsa|ssh-dss|ssh-ed25519|ecdsa-sha2-nistp256) /g;
```

Comment:
We have a validators file that this would make sense in.
Possibly just exporting a `validateSSHKey` function.
 https://github.com/evergreen-ci/spruce/blob/ddaa12cfb527caab3d8256c36eda659b97f37ed4/src/utils/validators/index.ts